WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -0,0 +1,135 @@
+import Combine
+import UIKit
+import Yosemite
+import protocol Storage.StorageManagerType
+
+/// Coordinates navigation for store creation flow, with the assumption that the app is already authenticated with a WPCOM user.
+final class StoreCreationCoordinator: Coordinator {
+    /// Navigation source to store creation.
+    enum Source {
+        /// Initiated from the login prologue in logged-out state.
+        case prologue
+        /// Initiated from the store picker in logged-in state.
+        case storePicker
+    }
+
+    var navigationController: UINavigationController
```

Comment:
❓ Is there any case where we want to update this from outside? Should we make it private instead?

Reply:
It's just to conform to `Coordinator` protocol that requires the navigation controller to be mutable:

https://github.com/woocommerce/woocommerce-ios/blob/34c040c9d156ea730c0d10a06c0e03554dfbe5d0/WooCommerce/Classes/Tools/Coordinator.swift#L8

I started an async discussion about this p1666768545000749-slack-CGPNUU63E, in the past I had the same question https://github.com/woocommerce/woocommerce-ios/pull/2669#discussion_r472885799 😄